specs/driver_spec.rb

```diff
@@ -1,104 +1,179 @@
+                                  rating: 1)
+      @driver.add_driven_trip(trip2)
+
+      trip3 = RideShare::Trip.new(id: 9, driver: @driver, passenger: nil,
+                                  start_time: Time.parse("2016-08-08"),
+                                  end_time: Time.parse("2016-08-09"), cost: 16,
+                                  rating: 1)
+      @driver.add_driven_trip(trip3)
     end
 
-    it "returns a float within range of 1.0 to 5.0" do
-      average = @driver.average_rating
-      expect(average).must_be :>=, 1.0
-      expect(average).must_be :<=, 5.0
+    it "correctly calculates total driver revenue" do
+    # You add tests for the total_revenue method
+      driver_revenue = @driver.total_revenue
+      expect (driver_revenue).must_equal (50 - ( 3 * 1.65 ) ) * 0.80
     end
 
-    it "returns zero if no trips" do
-      driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV",
-                                     vin: "1C9EVBRM0YBC564DZ")
-      expect(driver.average_rating).must_equal 0
+    it "ignores trips in progress when calculating total driver revenue" do
+      trip4 = RideShare::Trip.new(id: 10, driver: @driver, passenger: nil, start_time: Time.now, end_time: nil, cost: nil, rating: nil)
+      @driver.add_driven_trip(trip4)
+
+      driver_revenue = @driver.total_revenue
+      expect (driver_revenue).must_equal (50 - ( 3 * 1.65 ) ) * 0.80
     end
+  end
 
-    it "correctly calculates the average rating" do
-      trip2 = RideShare::Trip.new(id: 8, driver: @driver, passenger: nil,
-                                  date: Time.parse("2016-08-08"), rating: 1)
-      @driver.add_trip(trip2)
 
-      expect(@driver.average_rating).must_be_close_to (5.0 + 1.0) / 2.0, 0.01
-    end
+  describe "net_expenditures" do
+    before do
 
+      @user = RideShare::User.new(id: 54, name: "Rogers Bartell IV", phone: "353-533-5334")
 
-  end
 
-  describe "total_revenue" do
-    # You add tests for the total_revenue method
-  end
+      @driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV",
+                                      vin: "1C9EVBRM0YBC564DZ")
+      trip = RideShare::Trip.new(id: 8, driver: @driver, passenger: nil,
+                                 start_time: Time.parse("2016-08-08"),
+                                 end_time: Time.parse("2016-08-08"), cost: 4, rating: 5)
+      @driver.add_driven_trip(trip)
 
-  describe "net_expenditures" do
-    # You add tests for the net_expenditures method
+
+      trip = RideShare::Trip.new(id: 8, driver: 5, passenger: @driver,
+                                 start_time: Time.parse("2016-08-08"),
+                                 end_time: Time.parse("2016-08-08"), cost: 10, rating: 5)
+
+      @user.add_trip(trip)
+      it 'checks if driver.net_expenditures overrides user.net_expenditures' do
+      difference = @driver.net_expenditures
+      expect(difference).must_equal 10 - ((4 - 1.65) * 0.80)
+      end
```

Comment:
This is a good start, but I believe there are some more interesting cases here. For example, what if a driver has:

- No driven trips
- No trips as a passenger
- An incomplete trip as a driver or passenger
- Made more money than they've spent